tritium-metrics/src/test/java/com/palantir/tritium/metrics/TaggedMetricsThreadFactoryTest.java

```diff
@@ -44,8 +47,55 @@ void testInstrumentation() {
                 .build();
         ThreadFactory instrumented = MetricRegistries.instrument(registry, delegate, name);
         ExecutorMetrics metrics = ExecutorMetrics.of(registry);
-        Counter running = metrics.threadsRunning(name);
-        Meter created = metrics.threadsCreated(name);
+        Counter running = metrics.threadsRunning()
+                .executor(name)
+                .threadType(ThreadsRunning_ThreadType.PLATFORM)
+                .build();
+        Meter created = metrics.threadsCreated()
+                .executor(name)
+                .threadType(ThreadsCreated_ThreadType.PLATFORM)
+                .build();
+        assertThat(running.getCount()).isZero();
+        assertThat(created.getCount()).isZero();
+        CountDownLatch latch = new CountDownLatch(1);
+        Thread thread = instrumented.newThread(() -> Uninterruptibles.awaitUninterruptibly(latch));
+        assertThat(created.getCount()).isOne();
+        // thread has not started yet
+        assertThat(running.getCount()).isZero();
+        thread.start();
+        // Allow the thread to start in the background
+        Awaitility.waitAtMost(Duration.ofSeconds(3)).untilAsserted(() -> {
+            assertThat(created.getCount()).isOne();
+            assertThat(running.getCount()).isOne();
+        });
+        latch.countDown();
+        Awaitility.waitAtMost(Duration.ofSeconds(3)).untilAsserted(() -> {
+            assertThat(created.getCount()).isOne();
+            assertThat(running.getCount()).isZero();
+        });
+        Awaitility.waitAtMost(Duration.ofSeconds(1))
+                .untilAsserted(() -> assertThat(thread.isAlive()).isFalse());
+    }
+
+    @Test
+    void testVirtualThreadInstrumentation() {
```

Comment:
These tests share a fair bit of code. I bias away from sharing functionality between tests because it often results in refactors down the line that reduce the efficacy of some tests. In this case the tests are validating very similar things with different inputs, so there may be a stronger argument for sharing test code